### PR TITLE
Add Types

### PR DESCRIPTION
**Pull request by leonardoventurini**
image

### CODE REVIEW

File: types.d.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    stalledTimeout?: number
+}
+
+type TraceInfo = {
+    type?: string
+    name?: string
+}
+
+export namespace Tracer {
+    function addFilter(filterFunction: (eventType: string, data: any, info: TraceInfo) => any): void
```
[the remark this answers is not in the document]
Comment (leonardoventurini) [reply types.d.ts:20]: I guess in this case we should use `Record<string, any>`, which enforces a proper shape.

File: types.d.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    function startContinuousProfiling(): void;
+    function enableErrorTracking(): void;
+    function disableErrorTracking(): void;
+
+    function trackError(error: Error, options?: any): void;
+
+    function ignoreErrorTracking(error: Error): void;
+
+    function startEvent(name: string, data?: any): MontiEvent | false;
+    function endEvent(event: MontiEvent, data?: any): void;
```
[the remark this answers is not in the document]
Comment (leonardoventurini) [reply types.d.ts:37]: Same, we can use `Record<string, any>`

File: types.d.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    function startContinuousProfiling(): void;
+    function enableErrorTracking(): void;
+    function disableErrorTracking(): void;
+
+    function trackError(error: Error, options?: any): void;
+
+    function ignoreErrorTracking(error: Error): void;
+
+    function startEvent(name: string, data?: any): MontiEvent | false;
```
[the remark this answers is not in the document]
Comment (leonardoventurini) [reply types.d.ts:36]: Makes sense, will check

File: types.d.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    uploadSourceMaps?: boolean
+    recordIPAddress?: boolean
+    eventStackTrace?: boolean
+    disableNtp?: boolean
+    stalledTimeout?: number
+}
+
+type TraceInfo = {
+    type?: string
+    name?: string
```
[the remark this answers is not in the document]
Comment (leonardoventurini) [reply types.d.ts:16]: Nah, I think you are right. I made them optional because I wasn't sure.

File: types.d.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
+export type MontiEvent = any
+export type KadiraInfo = any
+
+export type ConnectOptions = {
+    enableErrorTracking?: boolean
+    endpoint?: string
+    hostname?: string
+    uploadSourceMaps?: boolean
+    recordIPAddress?: string
```
Comment (zodern) [inline types.d.ts:9]: Could we use a string union here with the allowed values?

File: types.d.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    eventStackTrace?: boolean
+    disableNtp?: boolean
+    stalledTimeout?: number
+}
+
+export type TrackErrorOptions = {
+    type?: string
```
Comment (zodern) [inline types.d.ts:16]: While any string is allowed, the UI only supports certain values. It probably would be good to list them here: `method`, `client`, `sub`, `server-crash`, `server-internal`

File: types.d.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    subType?: string
+    kadiraInfo?: KadiraInfo
+}
+
+export type TraceInfo = {
+    type: string
```
Comment (zodern) [inline types.d.ts:22]: There are only 3 values for the trace type: https://github.com/monti-apm/monti-apm-agent/blob/912d2623bc5961c2b6ce3665494edafe40072ba6/lib/tracer/tracer.js#L7

File: types.d.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    function startContinuousProfiling(): void;
+    function enableErrorTracking(): void;
+    function disableErrorTracking(): void;
+
+    function trackError(error: Error, options?: TrackErrorOptions): void;
+
+    function ignoreErrorTracking(error: Error): void;
+
+    function startEvent(name: string, data?: Record<string, any>): MontiEvent | false;
+    function endEvent(event: MontiEvent | false, data?: any): void;
```
Comment (zodern) [inline types.d.ts:46]: The type for this `data` should probably match the type for the data in `startEvent`

**Pull request merged**